CC/Sounder/main.cc

```diff
@@ -31,27 +32,41 @@ int main(int argc, char* argv[])
     gflags::ParseCommandLineFlags(&argc, &argv, true);
     Config config(
         FLAGS_conf, FLAGS_storepath, FLAGS_bs_only, FLAGS_client_only);
-    int ret = EXIT_SUCCESS;
+    int ret = EXIT_FAILURE;
     if (FLAGS_gen_ul_bits) {
         DataGenerator dg(&config);
         dg.GenerateData(FLAGS_storepath);
     } else {
-        try {
-            SignalHandler signalHandler;
+        int cnt = 0;
+        int maxTry = 2;
+        while (cnt++ < maxTry && ret == EXIT_FAILURE) {
+            try {
+                SignalHandler signalHandler;
 
-            // Register signal handler to handle kill signal
-            signalHandler.setupSignalHandlers();
-            config.loadULData(FLAGS_storepath);
-            Sounder::Recorder dr(&config);
-            dr.do_it();
-            ret = EXIT_SUCCESS;
-        } catch (SignalException& e) {
-            std::cerr << "SignalException: " << e.what() << std::endl;
-            ret = EXIT_FAILURE;
-        } catch (const std::exception& exc) {
-            std::cerr << "Program terminated Exception: " << exc.what()
-                      << std::endl;
-            ret = EXIT_FAILURE;
+                // Register signal handler to handle kill signal
+                signalHandler.setupSignalHandlers();
+                config.loadULData(FLAGS_storepath);
+                Sounder::Recorder dr(&config);
+                dr.do_it();
+                ret = EXIT_SUCCESS;
+
+            } catch (const SignalException& e) {
+                std::cerr << "SignalException: " << e.what() << std::endl;
+                ret = EXIT_FAILURE;
+                break;
+
+            } catch (const Sounder::RetryableError& rex) {
+                // Discovery usually fails on the first run, re-try
+                std::cout << "Exception: " << rex.what() << " Re-Try!"
+                          << std::endl;
+                usleep(1e6);
```

Comment:
```suggestion
                std::this_thread::sleep_for (std::chrono::seconds(1));
```
Why is the sleep necessary?

Reply:
not really necessary but helps to see what's happening during initialization/failure

Reply:
nevermind, it is necessary... need to wait a bit otherwise it will simply fail again if done right away